File: Baekjoon/Class3/03.1764 듣보잡.py
import sys
input = sys.stdin.readline

n, m = map(int, input().split())

unheard = set() #듣도 못한 사람
for _ in range(n):
    unheard.add(input().rstrip())

unseen = set() #보도 못한 사람
for _ in range(m):
    unseen.add(input().rstrip())

unheard_unseen = sorted(unheard & unseen) #듣도 보도 못한 사람
print(len(unheard_unseen))
print("\n".join(unheard_unseen))


'''
import sys
input = sys.stdin.readline

n, m = map(int, input().split())

unheard = set() #듣도 못한 사람
for _ in range(n):
    unheard.add(input().rstrip())

unheard_unseen = [] #듣도 보도 못한 사람
for _ in range(m):
    name = input().rstrip()
    if name in unheard:
        unheard_unseen.append(name)

unheard_unseen.sort() #사전순 정렬
print(len(unheard_unseen))
print("\n".join(unheard_unseen))
'''

File: Baekjoon/Class3/03.1764 듣보잡.js
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").trim().split("\n");

const [n, m] = input[0].split(" ").map(Number);

const unheard = new Set() // 듣도 못한 사람
for(let i = 1; i <= n; i++)
    unheard.add(input[i]);

const unheardUnseen = [] // 듣도 보도 못한 사람
for(let i = n + 1; i <= n + m; i++) {
    const name = input[i];
    if(unheard.has(name))
        unheardUnseen.push(name);
}

unheardUnseen.sort(); // 정렬
console.log(`${unheardUnseen.length}\n${unheardUnseen.join("\n")}`);